web/backend/src/controllers/users.ts

```diff
@@ -34,14 +34,28 @@ usersRouter.post('/add_role', (req, res, next) => {
     }
   }
 
-  addPolicy(req.body.userId, req.body.subject, req.body.permission)
-    .then(() => {
-      res.set(200).send();
-      next();
-    })
-    .catch((e) => {
-      res.status(400).send(`Error while adding to roles: ${e}`);
-    });
+  if (Object.hasOwn(req.body, 'userId')) {
+    addPolicy(req.body.userId, req.body.subject, req.body.permission)
+      .then(() => {
+        res.set(200).send();
+        next();
+      })
+      .catch((e) => {
+        res.status(400).send(`Error while adding to roles: ${e}`);
+      });
```

Comment:
You have to be careful with the `then` construct here, as it will continue executing the code below, which might not be what you want. If you think this is a problem (I didn't look at it in detail), you should change it to the following. Same for the `then` just below.

```
  try {
    await addPolicy(req.body.userId, req.body.subject, req.body.permission);
    res.set(200).send();
    next();
  } catch(e) {
    res.status(400).send(`Error while adding to roles: ${e}`);
  };
```